src/SearchHeuristics/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Document } from '../SearchIndex';
+
+// A series of basic heuristics that we can apply to determine if a search result should be indexed
+// TODO: Document interface should be defined here and this should be a document abstraction?
+// or a SearchableDocument abstraction?
+
+interface FilteredDocuments {
+  searchable: Document[];
+  unsearchable: Document[];
+}
+
+const hasMinimumTextThreshold = (doc: Document) => {
+  const MINIMUM_CONTENT_LENGTH = 40;
+  return doc.text.length >= MINIMUM_CONTENT_LENGTH;
+};
+
+const hasPreview = (doc: Document) => !!doc.preview?.trim();
+
+const isInteractive = (doc: Document) => doc.links?.length >= 2;
```

Comment:
Why do we require there to be 2 or more links in a document? Do we want a document to fail the threshold if it has 0 or 1 link but has content?

Reply:
It's an arbitrary metric I'm using to avoid false negatives in conjunction with text length and preview existence - the majority of our content has at least 1 or 2 links within a page, so on the off chance that a document's text didn't get properly inputted into the document by `mut`, this boolean can be used to prevent false negatives.

It's obviously a really low bar to clear, but I'm intending to err on the side of avoiding false negatives more than letting some drawer results slip through.

Reply:
Ohhh, that makes more sense. Thank you for the explanation!